fed/mosaic-sumo/src/main/java/org/eclipse/mosaic/fed/sumo/bridge/facades/SimulationFacade.java

```diff
@@ -596,13 +643,18 @@ private VehicleSensors createSensorData(SumoVehicleState vehicleState, LeadingVe
         double leaderSpeed = hasFrontSensorActivated ? Double.POSITIVE_INFINITY : -1;
         double rearDistance = hasBackSensorActivated ? Double.POSITIVE_INFINITY : -1;
 
-        if (hasFrontSensorActivated && leadingVehicle != LeadingVehicle.NO_LEADER) {
-            frontDistance = leadingVehicle.getLeadingVehicleDistance() + minGap;
-            VehicleData leadingVehicleData = getLastKnownVehicleData(leadingVehicle.getLeadingVehicleId());
+        if (hasFrontSensorActivated
+                && leadingVehicle.getDistance() < vehicleState.frontSensorDistance
+        ) {
+            frontDistance = leadingVehicle.getDistance() + minGap;
+            VehicleData leadingVehicleData = getLastKnownVehicleData(leadingVehicle.getOtherVehicleId());
             leaderSpeed = leadingVehicleData != null ? leadingVehicleData.getSpeed() : -1;
         }
-        if (hasBackSensorActivated && followerDistance != null) {
-            rearDistance = followerDistance;
+        if (hasBackSensorActivated
+                && followerVehicle.getDistance() < vehicleState.rearSensorDistance
+        ) {
+            // using the own minGap is not 100% correct here, but it would be very expensive to fetch the minGap of the follower
+            rearDistance = followerVehicle.getDistance() + minGap;
```

Comment:
not only is `minGap` inaccurate but wouldn't it make sense to measure from the back-bumper?